src/integrationTest/groovy/wooga/gradle/fastlane/FastlaneIntegrationSpec.groovy

```diff
@@ -16,11 +16,73 @@
 
 package wooga.gradle.fastlane
 
-abstract class FastlaneIntegrationSpec extends FastlaneSpec {
+import com.wooga.gradle.PlatformUtils
+
+import java.nio.file.Paths
+
+abstract class FastlaneIntegrationSpec extends IntegrationSpec {
+
+    File fastlaneMock
+    File fastlaneMockPath
+
+    def setupFastlaneMock() {
+        fastlaneMockPath = File.createTempDir("fastlane", "mock")
+
+        def path = System.getenv("PATH")
+        environmentVariables.clear("PATH")
+        String newPath = "${fastlaneMockPath}${File.pathSeparator}${path}"
+        environmentVariables.set("PATH", newPath)
+        assert System.getenv("PATH") == newPath
+
+
+        fastlaneMock = createFile("fastlane", fastlaneMockPath)
+        fastlaneMock.executable = true
+        fastlaneMock << """
+            #!/usr/bin/env bash
+            echo \$@
+            env
+        """.stripIndent()
+    }
+
     def setup() {
+        setupFastlaneMock()
         buildFile << """
-          group = 'test'
-          ${applyPlugin(FastlanePlugin)}
-       """.stripIndent()
+              group = 'test'
+              ${applyPlugin(FastlanePlugin)}
+           """.stripIndent()
+    }
+
+
+
+    // TODO: Replace with newer test API
+    // Should not use project dir if on not windows
+    Object substitutePath(Object str, Object subStr, String typeName) {
+
+        if (typeName != "File" && typeName != "Provider<RegularFile>") {
+            return str
+        }
+
+        def path = (String) subStr
+        if (path == null) {
+            return str
+        }
+
+        // If it's an absolute path starting from the current volume
+        if (Paths.get(path).isAbsolute()){
+            return str
+        }
+//        if (!PlatformUtils.windows && path.startsWith("/")){
+//            return str
+//        }
+//        else if (PlatformUtils.windows && path.startsWith("c:")) {
+//            return str
+//        }
+
+
+        def modifiedPath = typeName == "Provider<RegularFile>"
+            ? "/build${path}"
+            : path
+
+        str.replace(path, new File(projectDir, modifiedPath).path)
```

Comment:
I don't get this part at all. Or better the whole method for that matter. 
So `subStr` which is an Object get renamed and casted to `path`.
The variable `str` which is also of type `Object` will be used when the type is not `Provider<RegularFile>` here you call `replace` which I guess is a String API. You always set the absolute path for `Provider<RegularFile>` to be `projectDir/build`. All in all this method is super confusing. You also seem to expect that `subStr` starts with a `/`. I didn't checked the usage here. But please don't expect to be replaced path objects to be absolute and then append them as relative parts. A natural join should look like `/build/${path}`.

Reply:
I agree with the method being confusing; I wrote it as a stopgap measure to fix the tests until I could switch over to a newer API that deals with this gracefully. 

I will try to refactor it.